docs/interpreter_design_doc.md

```diff
@@ -0,0 +1,100 @@
+# Interpreter Design
+
+
+## Class Hierarchy
+
+Stablehlo programs are computations over Tensors (n-dimensional arrays), hence
```

Comment:
Do we have to capitalize "Tensor" here? It's a common noun, so it probably should be written as "tensor"?